### PR TITLE
Split liquibase-core and liquibase-commercial jar files

### PR DESCRIPTION
**Pull request by nvoxland**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

Rather than having liquibase-core contain both the functionality from this repository as well as the commercial code, keep "liquibase-core" as the base, open source library and create a new "liquibase-commercial" as the additional proprietary/commercial functionality.

The new liquibase-commercial.jar file will be available:
- In maven central
- As an additional asset on release pages
- As an additional jar in the `liquibase-artifacts.zip` and `liquibase-artifacts-<BRANCH_NAME>.zip` build artifacts

For consistency, the liquibase-core.jar file will now be stored as `liquibase-core.jar` on the build and release assets pages rather than `liquibase.jar` like it previously was.

### CLI

The liquibase CLI distributions will continue to include both `liquibase-core.jar` and `liquibase-commerical.jar` in the `internal/lib` directory**_, so there is no impact for CLI users,_** regardless of what distribution is used (tar.gz file, zip file, or docker image). The existing liquibase launch scripts have been updated to use the correct jar files.

While we will not ship a CLI distribution without `liquibase-commercial.jar`, removing any non-open source code is simply a matter of deleting `internal/lib/liquibase-commercial.jar`

**_NOTE_**: The `liquibase-core.jar` file was previously called `liquibase.jar` and stored in the root of LIQUIBASE_HOME. So anyone working directly with the `liquibase.jar` file will need to update their usage to use `internal/lib/liquibase-core.jar`.

**_NOTE_**: If you upgrade over an existing LIQUIBASE_HOME installation, the old liquibase.jar file will continue to exist because there is no logic to clean up old files. It can be deleted at your leisure, and it will not be used by Liquibase if left.

### Maven Plugin

The Liquibase-maven-plugin will now depend on both `liquibase-core` and `liquibase-commercial` instead of just `liquibase-core`. This means that anyone using the maven plugin as:
```
            <plugin>
                <groupId>org.liquibase</groupId>
                <artifactId>liquibase-maven-plugin</artifactId>
                ...
```
will continue to be able to use both free and paid functionality with **_no changes to their configuration_**.

### Servlet / Spring / Ant Integrations

See "User Managed Dependencies" below

### User Managed Dependencies

Any projects that were directly or indirectly referencing or using "liquibase-core" will no longer include the commercial functionality. **_To continue to use Liquibase Pro features, you need to have both liquibase-core.jar and liquibase-commercial.jar as part of your project._**

If you are managing your project using a tools that follows Maven managed dependencies (Maven, Gradle, Ivy, etc.), the easiest approach is to change your dependency on `liquibase-core` to `liquibase-commercial`. Because liquibase-commercial defines liquibase-core as a dependency, you will automatically get both included in your project without having to manage multiple dependencies yourself. 

If you are not directly referencing liquibase-core yourself, but it is comes along as a transitive dependency (like if you use spring-boot), add `org.liquibase:liquibase-commercial` as an additional dependency in your project.

### CODE REVIEW

**Comment by nvoxland** [comment]
Things to test:
- In the build's liquibase-artifacts-*.zip artifacts:
  - Both liquibase-core.jar and liquibase-commercial.jar exist
- The liquibase-core.jar:
  -  Does not include a `liquibase/pro` directory
- The liquibase-maven-plugin.jar's:
  -  META-INF/maven/.../pom.xml file has both liquibase-core and liquibase-commercial as dependnecies
- The liquibase-commercial-jar:
  - Has liquibase-core as a dependency
  - Has correct license and other metadata
  - Does not include any other dependencies

**Comment by nvoxland** [comment]
I think this is all working now, including the "re-version" logic. I tested that with a test run of the action.

**Review by suryaaki2 — APPROVED** [review]

File: liquibase-core/src/main/resources/liquibase/examples/start-h2.bat [hunk 1]
```diff
@@ -6,7 +6,7 @@ setlocal enabledelayedexpansion
 rem %~dp0 is expanded pathname of the current script under NT
 rem %~p0 is the directory of the current script
 
-if exist %~p0\..\liquibase.jar SET LIQUIBASE_HOME=%~p0..
+if exist %~p0\..\GETTING_STARTED.txt SET LIQUIBASE_HOME=%~p0..
```
[the remark this answers is not in the document]
Comment (nvoxland) [reply liquibase-core/src/main/resources/liquibase/examples/start-h2.bat]: Yeah, not super robust. But this is also a legacy "nobody should be using it" file anyway...

Nevertheless, I switched it to liquibase.bat as a file that is maybe less likely to be randomly deleted by someone.

**Comment by FBurguer** [comment]
TEST:

1. MacOS Installer:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key: `PASS`
2. Windows Installer:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key: `PASS`
3. Ant Integration:
- Pro Features access denied without Pro Key and with `liquibase-commercial.jar` included: `PASS`
- Pro Features access with Pro Key and with `liquibase-commercial.jar` included: `PASS`
- Pro Features unavailble with Pro Key but without `liquibase-commercial.jar` included: `PASS`
- Upgrade from core to Pro with `liquibase-commercial.jar` included and Pro Key: `PASS`
4. Servlet Integration:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key (not available at the moment but the code is accesable): `PASS`
- Pro Features unavailble with Pro Key but without `liquibase-commercial.jar` included: `PASS`
- Upgrade from core to Pro with `liquibase-commercial.jar` included and Pro Key(not available at the moment but the code is accesable): `PASS`
5. .zip file:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key: `PASS`
6. .tar file:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key: `PASS`
7. Spring-boot/Spring integration:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key (not available at the moment but the code is accesable): `PASS`
- Pro Features unavailble with Pro Key but without `liquibase-commercial.jar` included: `PASS`
- Upgrade from core to Pro with `liquibase-commercial.jar` included and Pro Key(not available at the moment but the code is accesable): `PASS`
8. Maven plug-in:
- Pro Features access denied without Pro Key: `PASS`
- Pro Features access with Pro Key: `PASS`
(there is no way to exclude `liquibase-commercial.jar` at the momment)

**Comment by FBurguer** [comment]
Post Release distributions pending:

- Gradle Integration
- liquibase.org/download
- Hombrew
- Choclatery
- Docker

**Comment by kataggart** [comment]
this PR makes me so happy
that is all

**Pull request merged**